avocado/core/messages.py

```diff
@@ -172,7 +172,7 @@ class LogMessageHandler(BaseRunningMessageHandler):
     """
 
     def handle(self, message, task, job):
-        self._save_message_to_file('debug.log', message['log'], task)
+        self._save_message_to_file('debug.log', "%s\n" % message['log'], task)
```

Comment:
> This sounds strange to me. I would add add new argument, something like:
> 
> ```python
>   def _save_message_to_file(filename, buff, task, mode='a', newline=False):
>         file = os.path.join(task.metadata['task_path'], filename)
>         with open(file, mode) as fp:
>             if newline:
>                 buff = "%s\n" % buff
>             fp.write(buff)
> ```

I thought of doing that, but to me, the addition of a newline is a very special case of the `LogMessageHandler` that should be limited to it.  We've had issues in the past where we made it to easy to add newlines to stdout/stderr lines and problems slipped in.

I think the data should be saved without any manipulation, and TBH, only because I wanted to fix the issue without changing more of the behavior that I kept adding the newline, but I'd really like to revisit/reconsider that.

Reply:
I see, we could at least left a comment on LogMessageHandler, why this is being added.

Reply:
The commit message explains it:

```
The current implementation assumes that the content of the "log"
messages will contain text, but that's not always the case.  This
moves the newline handling so that others can choose to not do that.
```

Do you think it needs to be on the code too?